ugvc/pipelines/lpr/train_lib_prep_recalibration_model.py

```diff
@@ -31,49 +33,103 @@ def run(argv):
+            [],
+            {
+                "-i": calls_vcf,
+                "-r": args.ref_fasta,
+                "-f": "TGCA",
+                "--info_fields_override": "",
+                "--format-fields": "DP=int AD=str SB=str VAF=str PL=str BG_AD=str",
+                "-o": calls_parquet,
+            },
+            flags=["--report-bases-in-reference-direction"],
         )
 
-    for bed_file, vcf_file, parquet_file, min_af, max_af in (tp_args, fp_args):
-        sp.print_and_run(f"awk '$4 >= {min_af} && $4 <= {max_af}' {afs_bed} > {bed_file}")
         sp.print_and_run(
-            f"bedtools intersect -a {featuremap_vcf} -b {bed_file} -u -header | " f"bcftools view -Oz -o {vcf_file}"
+            f"papermill {nb_dir}/qual_af_clustering.ipynb "
+            f"{out_dir}/qual_af_clustering.nbconvert.ipynb "
+            f"-p vcf_input {calls_parquet} "
+            f"-p output_file {qual_af_clusters_on_calls} "
+        )
+        sp.print_and_run(
+            "jupyter nbconvert --to html "
+            f"{out_dir}/qual_af_clustering.nbconvert.ipynb "
+            f"--no-input --output qual_af_clustering.html"
         )
         sp.print_and_run_clt(
-            featuremap_to_dataframe.run, [], {"-i": vcf_file, "-r": args.ref_fasta, "-f": "TGCA", "-o": parquet_file}
+            featuremap_to_dataframe.run,
+            [],
+            {"-i": featuremap_on_calls_vcf, "-r": args.ref_fasta, "-f": "TGCA", "-o": featuremap_on_calls_parquet},
+        )
+        qual_af_df = pd.read_parquet(qual_af_clusters_on_calls)
+        featuremap_on_calls_df = pd.read_parquet(featuremap_on_calls_parquet)
+        featuremap_on_calls_df = featuremap_on_calls_df.drop(columns=["qual", "filter"])
+        qual_af_df.merge(featuremap_on_calls_df, on=["chrom", "pos", "ref", "alt"]).to_parquet(labeled_training_set)
+        del qual_af_df
+        del featuremap_on_calls_df
+        gc.collect()
+
+    # calibrate model on raw-data
+    else:
+        sp.print_and_run(
+            f"bcftools query -f '%CHROM\t%POS0\t%POS\t%X_READ_COUNT' {featuremap_vcf} "
+            " | bedtools groupby -c 3 -full -o count | awk -v OFS='\t' '{print $1,$2,$3,$5/$4}' " + f" > {afs_bed}"
         )
 
-    base_dir = args.notebooks_dir
+        for bed_file, vcf_file, parquet_file, min_af, max_af in (tp_args, fp_args):
+            sp.print_and_run(f"awk '$4 >= {min_af} && $4 <= {max_af}' {afs_bed} > {bed_file}")
+            sp.print_and_run(
+                f"bedtools intersect -a {featuremap_vcf} -b {bed_file} -u -header | " f"bcftools view -Oz -o {vcf_file}"
+            )
+            sp.print_and_run_clt(
+                featuremap_to_dataframe.run,
+                [],
+                {"-i": vcf_file, "-r": args.ref_fasta, "-f": "TGCA", "-o": parquet_file},
+            )
+        fp_df = pd.read_parquet(fp_args[2])
+        tp_df = pd.read_parquet(tp_args[2])
+        fp_df["label"] = False
+        tp_df["label"] = True
```

Comment:
Doesn't it warn you to do this instead?
```suggestion
        tp_df.loc[:, "label"] = True
```

Reply:
Probably does, will try to fix.